homeassistant/components/ihc/__init__.py

```diff
@@ -0,0 +1,122 @@
+"""IHC platform."""
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+import time
+import logging
+import os.path
+import xml.etree.ElementTree
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_URL, CONF_USERNAME, CONF_PASSWORD
+from homeassistant.config import load_yaml_config_file
+
+from homeassistant.components.ihc import const
+
+REQUIREMENTS = ['ihcsdk==2.1.0']
+DOMAIN = 'ihc'
+
+CONF_INFO = 'info'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_INFO): cv.boolean
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """Setyp the IHC platform."""
+    from ihcsdk.ihccontroller import IHCController
+    url = config[DOMAIN].get(CONF_URL)
+    username = config[DOMAIN].get(CONF_USERNAME)
+    password = config[DOMAIN].get(CONF_PASSWORD)
+    ihc = IHCController(url, username, password)
+    ihc.info = config[DOMAIN].get(CONF_INFO)
+
+    if not ihc.authenticate():
+        _LOGGER.error("Unable to authenticate on ihc controller.")
+        return False
+
+    hass.data[DOMAIN] = IHCPlatform(ihc)
+
+    # Service functions
+
+    def set_runtime_value_bool(call):
+        """Set a IHC runtime bool value service function."""
+        ihcid = int(call.data.get('ihcid', 0))
+        value = bool(call.data.get('value', 0))
+        ihc.set_runtime_value_bool(ihcid, value)
+
+    def set_runtime_value_int(call):
```

Comment:
I'm thinking, instead of creating three services, maybe one service is enough that takes a type in the service data. But that might just make things more complex. You decide.

Reply:
It is important to have the type correct when your set a runtime value on the IHC controller. For example if you try to set a float value with an int, it will fail. So I prefer having separate functions for each type. And now when service data is validated (see below). it is even better.